alerting/src/main/kotlin/org/opensearch/alerting/TriggerService.kt

```diff
@@ -60,31 +63,18 @@ class TriggerService(val scriptService: ScriptService) {
     fun runDocLevelTrigger(
         monitor: Monitor,
         trigger: DocumentLevelTrigger,
-        ctx: DocumentLevelTriggerExecutionContext,
-        docsToQueries: Map<String, List<String>>,
-        queryIds: List<String>
+        queryToDocIds: Map<DocLevelQuery, Set<String>>
     ): DocumentLevelTriggerRunResult {
         return try {
-            val triggeredDocs = mutableListOf<String>()
+            var triggeredDocs = mutableListOf<String>()
 
-            val dummyTrigger = QueryLevelTrigger(
-                name = trigger.name,
-                severity = trigger.severity,
-                actions = trigger.actions,
-                condition = trigger.condition
-            )
-            val dummyExecutionContext = QueryLevelTriggerExecutionContext(monitor, dummyTrigger, emptyList(), Instant.now(), Instant.now())
-
-            for (doc in docsToQueries.keys) {
-                val params = trigger.condition.params.toMutableMap()
-                for (queryId in queryIds) {
-                    params[queryId] = docsToQueries[doc]!!.contains(queryId)
+            if (trigger.condition.idOrCode.equals(ALWAYS_RUN.idOrCode)) {
+                for (value in queryToDocIds.values) {
+                    triggeredDocs.addAll(value)
                 }
-                val triggered = scriptService.compile(trigger.condition, TriggerScript.CONTEXT)
-                    .newInstance(params)
-                    .execute(dummyExecutionContext)
-                logger.info("trigger val: $triggered")
-                if (triggered) triggeredDocs.add(doc)
+            } else if (!trigger.condition.idOrCode.equals(NEVER_RUN.idOrCode)) {
+                triggeredDocs = TriggerExpressionParser(trigger.condition.idOrCode).parse()
+                    .evaluate(queryToDocIds).toMutableList()
```

Comment:
the `if-else if` block moves the `object equals check` to runtime. Kotlin allows this check to be made at compile time using `when` control statements.

```
when (trigger.condition.idOrCode) {
                ALWAYS_RUN.idOrCode  -> {
                    for (value in queryToDocIds.values) {
                        triggeredDocs.addAll(value)
                    }
                }
                NEVER_RUN.idOrCode -> {
                    triggeredDocs = TriggerExpressionParser(trigger.condition.idOrCode).parse()
                        .evaluate(queryToDocIds).toMutableList()
                }
            }
```
this will allow catching runtime errors we faced today at compile time only.

Reply:
Thanks for the suggestion @sbcd90. Agree, control statement could make the flow look much cleaner and I have used it in the Parser/Evaluation logic. However, here since this is not an instance check, but rather the value check with string comparison, I am not sure having control statement would add any value here in terms of performance. Given string value comparison still need to be performed at runtime.

Since we weren't using all the three possible values here, I chose to keep it as if-else-of block instead.